crates/asset-importer-rs-post-process/src/steps/flip_uvs.rs

```diff
@@ -1,20 +1,203 @@
 use asset_importer_rs_core::{AiPostProcess, AiPostProcessSteps};
-use asset_importer_rs_scene::AiScene;
+use asset_importer_rs_scene::{
+    AiMaterial, AiMesh, AiPropertyTypeInfo, AiReal, AiScene, AiUvTransform, AiVector3D, matkey,
+};
+use bytemuck;
 use enumflags2::BitFlags;
 
+#[derive(Debug, PartialEq)]
+pub enum FlipUVsError {
+    UvsNotClippedError(String, usize),
+}
+impl std::fmt::Display for FlipUVsError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            FlipUVsError::UvsNotClippedError(channel_name, channel_index) => {
+                write!(
+                    f,
+                    "UVs in channel named ({}) and with channel index ({}) do not have all their uv components between 0.0 and 1.0 ",
+                    channel_name, channel_index
+                )
+            }
+        }
+    }
+}
+impl std::error::Error for FlipUVsError {}
+
+#[derive(Copy, Clone)]
+pub enum UvFlipVariant {
+    X,
+    Y,
+    Z,
+}
+impl UvFlipVariant {
+    pub fn apply_flip_material(&self, uv_transform: &mut AiUvTransform) {
+        match self {
+            UvFlipVariant::X => {
+                uv_transform.translation.x *= -1.0;
+                uv_transform.rotation *= -1.0;
+            }
+            UvFlipVariant::Y => {
+                uv_transform.translation.y *= -1.0;
+                uv_transform.rotation *= -1.0;
+            }
+            UvFlipVariant::Z => {
+                uv_transform.translation.z *= -1.0;
+                uv_transform.rotation *= -1.0;
+            }
+        }
```

Comment:
```suggestion
    pub fn apply_flip_material(&self, uv_transform: &mut AiUvTransform) {
        match self {
            UvFlipVariant::X => {
                uv_transform.translation.x *= -1.0;
                uv_transform.rotation = AI_MATH_PI - uv_transform.rotation;
            }
            UvFlipVariant::Y => {
                uv_transform.translation.y *= -1.0;
                uv_transform.rotation *= -1.0;
            }
            UvFlipVariant::Z => {
                uv_transform.translation.z *= -1.0;
            }
        }
```

I believe the proper behavior to be this.

Reply:
This makes more sense. 👍